hack/generator/cmd/gen/gen.go

```diff
@@ -58,55 +58,20 @@ func NewGenCommand() (*cobra.Command, error) {
 			scanner := jsonast.NewSchemaScanner(idfactory)
 			scanner.AddFilters(viper.GetStringSlice("resources"))
 
-			_, err = scanner.ToNodes(ctx, root)
+			definitions, err := scanner.GenerateDefinitions(ctx, root)
 			if err != nil {
 				log.Printf("Error: %v\n", err)
 				return err
 			}
 
-			err = os.RemoveAll(rootOutputDir)
+			err = prepareOutputDirectory(rootOutputDir)
 			if err != nil {
-				log.Printf("Error: %v\n", err)
-				return err
-			}
-
-			err = os.Mkdir(rootOutputDir, 0700)
-			if err != nil {
-				log.Printf("Error %v\n", err)
 				return err
 			}
 
 			log.Printf("INF Checkpoint\n")
 
-			// group definitions by package
```

Comment:
Some of this may conflict with Bevan's PR #102

Reply:
Rebased on that